app/code/Magento/Catalog/Test/Unit/Model/ProductRepositoryTest.php

```diff
@@ -313,23 +313,37 @@ public function testGetProductInEditMode()
             ->will($this->returnValue('test_id'));
         $this->productMock->expects($this->once())->method('setData')->with('_edit_mode', true);
         $this->productMock->expects($this->once())->method('load')->with('test_id');
-        $this->productMock->expects($this->once())->method('getSku')->willReturn($sku);
+        $this->productMock->expects($this->once())->method('getId')->willReturn('test_id');
         $this->assertEquals($this->productMock, $this->model->get($sku, true));
     }
 
     public function testGetBySkuWithSpace()
     {
-        $trimmedSku = 'test_sku';
         $sku = 'test_sku ';
         $this->productFactoryMock->expects($this->once())->method('create')
             ->will($this->returnValue($this->productMock));
         $this->resourceModelMock->expects($this->once())->method('getIdBySku')->with($sku)
             ->will($this->returnValue('test_id'));
         $this->productMock->expects($this->once())->method('load')->with('test_id');
-        $this->productMock->expects($this->once())->method('getSku')->willReturn($trimmedSku);
+        $this->productMock->expects($this->once())->method('getId')->willReturn('test_id');
         $this->assertEquals($this->productMock, $this->model->get($sku));
     }
 
+    /**
+     * Test we can get product using get() with different cased SKU
+     */
+    public function testGetBySkuWithDifferentCase()
+    {
+        $differentCasedSku = 'tEsT_sKU';
+        $this->productFactoryMock->expects($this->once())->method('create')
+            ->will($this->returnValue($this->productMock));
+        $this->resourceModelMock->expects($this->once())->method('getIdBySku')->with($differentCasedSku)
+            ->will($this->returnValue('test_id'));
+        $this->productMock->expects($this->once())->method('load')->with('test_id');
+        $this->productMock->expects($this->once())->method('getId')->willReturn('test_id');
```

Comment:
Same here: it looks like we don't check getting product by `'test_sku'` anymore.